Library/Homebrew/cmd/fix-permissions.rb

```diff
@@ -0,0 +1,142 @@
+module Homebrew
+  def fix_permissions
+    checker = PermissionChecker.load_from_local
+    if ARGV.force?
+      begin
+        checker.fix!
+      ensure
+        system "/usr/bin/sudo", "-k"
+      end
+    else
+      return if checker.check!
+
+      unless checker.chmods.empty?
+        opoo "The following directories should be made group writable:"
+        puts(checker.chmods)
+      end
+
+      unless checker.chowns.empty?
+        opoo "The following directories should have their owner set to #{Tty.underline 39}#{ENV['USER']}#{Tty.reset}:"
+        puts(checker.chowns)
+      end
+
+      unless checker.chgrps.empty?
+        opoo "The following directories should have their group set to #{Tty.underline 39}admin#{Tty.reset}:"
+        puts(checker.chgrps)
+      end
+
+      unless HOMEBREW_CACHE.directory?
+        opoo "#{HOMEBREW_CACHE} doesn't exist."
+      end
+
+      puts "To fix permissions, run `brew fix-permissions --force`"
+      Homebrew.failed = true
+    end
+  end
+end
+
+class PermissionChecker
+  attr_reader :paths
+  attr_reader :chmods, :chowns, :chgrps
+
+  def self.load_from_local
+    load_from_yaml (HOMEBREW_LIBRARY_PATH/"permission_checklist.yml").read
+  end
+
+  # only used in install script
+  def self.load_from_remote
+    url = "https://raw.githubusercontent.com/Homebrew/homebrew/master/Library/Homebrew/permission_checklist.yml"
+    remote_yaml = `curl -fsSL ${url}`
+    raise "Fail to fetch remote resource: #{url}" if remote_yaml.empty?
+    load_from_yaml remote_yaml
+  end
```

Comment:
> I'd like to see how the install script looks as presumably the YAML will need embedded in that script too. 

@mikemcquaid This is how install script will load yaml file. So we only need to maintain one checklist.